internal/resource/config/oauth/accesstokenmanagers/settings/oauth_access_token_manager_settings_resource_gen.go

```diff
@@ -0,0 +1,193 @@
+// Code generated by ping-terraform-plugin-framework-generator
+
+package oauthaccesstokenmanagerssettings
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	client "github.com/pingidentity/pingfederate-go-client/v1210/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+var (
+	_ resource.Resource                = &oauthAccessTokenManagerSettingsResource{}
+	_ resource.ResourceWithConfigure   = &oauthAccessTokenManagerSettingsResource{}
+	_ resource.ResourceWithImportState = &oauthAccessTokenManagerSettingsResource{}
+)
+
+func OauthAccessTokenManagerSettingsResource() resource.Resource {
+	return &oauthAccessTokenManagerSettingsResource{}
+}
+
+type oauthAccessTokenManagerSettingsResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+func (r *oauthAccessTokenManagerSettingsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_oauth_access_token_manager_settings"
+}
+
+func (r *oauthAccessTokenManagerSettingsResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+}
+
+type oauthAccessTokenManagerSettingsResourceModel struct {
+	DefaultAccessTokenManagerRef types.Object `tfsdk:"default_access_token_manager_ref"`
+}
+
+func (r *oauthAccessTokenManagerSettingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"default_access_token_manager_ref": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"id": schema.StringAttribute{
+						Required:    true,
+						Description: "The ID of the resource.",
+					},
+				},
+				Optional:    true,
```

Comment:
It might actually make sense for us to just make this Required in the provider, since it is the only thing you would be using this resource for anyways.

Reply:
done